spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala

```diff
@@ -45,7 +41,7 @@ class HilbertSpaceTimeKeyIndex(
   val timeWidth = keyBounds.maxKey.temporalKey.time.getMillis - startMillis
   val temporalBinCount = math.pow(2, temporalResolution)
  
-  val chc = {
+  lazy val chc = {
```

Comment:
I believe that `lazy` is exactly doing what you advise me, or at least it once did.

> Le 17 août 2015 à 17:30, Eugene Cheipesh <notifications@github.com> a écrit :
> 
> In spark/src/main/scala/geotrellis/spark/io/index/hilbert/HilbertSpaceTimeKeyIndex.scala <https://github.com/geotrellis/geotrellis/pull/1159#discussion_r37199853>:
> 
> > @@ -45,7 +41,7 @@ class HilbertSpaceTimeKeyIndex(
> >    val timeWidth = keyBounds.maxKey.temporalKey.time.getMillis - startMillis
> >    val temporalBinCount = math.pow(2, temporalResolution)
> >   
> > -  val chc = {
> > +  lazy val chc = {
> Good to know that java serialization does not chase after lazy vals.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/geotrellis/geotrellis/pull/1159/files#r37199853>.
>